router/core/src/main/scala/io/buoyant/router/RoutingFactory.scala

```diff
@@ -53,6 +53,8 @@ object RoutingFactory {
 
   }
 
+  abstract class ResponseException[Rsp](rsp: Rsp) extends NoStackTrace
```

Comment:
I think this can just be a trait or abstract class with no type param or constructor args.  The child class will contain the response anyway and we don't have useful information about the type at this level.